src/edu/csus/ecs/pc2/clics/API202306/JudgementService.java

```diff
@@ -0,0 +1,289 @@
+                    }
+                }
+
+                exceptProps.clear();
+                cJudgment = new CLICSJudgement(model, controller, run, exceptProps);
+                // Remove max_run_time property if indicated
+                if(!includeMaxRunTime) {
+                    exceptProps.add("max_run_time");
+                }
+                try {
+                    // for this judgment, create filter to omit unused/bad properties (max_run_time in this case)
+                    SimpleBeanPropertyFilter filter = SimpleBeanPropertyFilter.serializeAllExcept(exceptProps);
+                    FilterProvider fp = new SimpleFilterProvider().addFilter("rtFilter", filter).setFailOnUnknownId(false);
+                    // generate json with only properties we want and add to CSV list.
+                    allJudgments.add(mapper.writer(fp).writeValueAsString(cJudgment));
+                } catch (Exception e) {
+                    return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Error creating JSON for judgment " + run.getElementId().toString() + " " + e.getMessage()).build();
+                }
+            }
+        }
+        return Response.ok("[" + allJudgments.toString() + "]", MediaType.APPLICATION_JSON).build();
+    }
+
+    /**
+     * Returns a representation of a specified judgment for the specified contest in JSON format. The returned value compliant with 2023-06
+     *
+     * @param sc User's infor
+     * @param contestId The contest
+     * @param judgementId The judgement we're looking for
+     * @return response
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{judgementId}/")
+    public Response getJudgement(@Context SecurityContext sc, @PathParam("contestId") String contestId, @PathParam("judgementId") String judgementId) {
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == true) {
+           long freezeTime = Utilities.getFreezeTime(model);
+           Run.RunStates status;
+           HashMap<String,HashMap<ElementId,Boolean>> allTeamsSolvedMap = null;
+           HashMap<ElementId,Boolean> teamSolvedMap;
+           String user = sc.getUserPrincipal().getName();
+           String runUser;
+           boolean isAdminUser = sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) || sc.isUserInRole(WebServer.WEBAPI_ROLE_JUDGE);
+           // admins can always see max_run_time
+           boolean includeMaxRunTime = isAdminUser;
+
+           for(Run run: model.getRuns()) {
+               // judgementId's match runId's - we found the one we are looking for
+               if (run.getElementId().toString().equals(judgementId)) {
+                   status = run.getStatus();
+
+                   // Check if judged or being judged - can't generate a judgment if not one of these states
+                   if(run.isJudged() || (status == RunStates.BEING_JUDGED || status == RunStates.BEING_COMPUTER_JUDGED)) {
```

Comment:
Same comment as under method `getJudgements()`:  I don't see how we can return a judgement for a run whose status is "BEING_JUDGED" (which by definition would seem to mean "has not yet completed judging"...)

Reply:
See above response about spec says to not include `judgement_type_id `for unjudged runs.